### ISSUE TITLE
1.0.35 crashing in Safari (iPhone)

### ISSUE THREAD

**Issue by schwarzjr** [d07RiV/diabloweb#79]
Worked before, but unfortunately don't know which version it was. Tried today morning again and 1. doesn't show the Blizzard logo at the beginning and 2. crashes after creating new/loading existing character. Just shows black screen with the touch-icons and then reloads the page after a couple of second.

**Comment by d07RiV** [comment]
How long ago did it work? Recently I only changed multiplayer and error handling, not sure how either of those could be related to videos not showing.

**Comment by schwarzjr** [comment]
Is it possible it worked yesterday evening? I downloaded diabdat.mpq on the phone and tried it. I think it did run, but maybe that was some older cached version. I didn't check the version number. I can try running the older versions locally in the evening and let you know. If I find a way how to do it :)

**Comment by schwarzjr** [comment]
OK, so the last working version is 1.0.29. Had to revert the f0ba5c7 commit in gh-pages branch for it to run on ios safari again. Maybe the "oops fix" in the master branch?

**Comment by d07RiV** [comment]
"Oops fix" was fixing an accidental commit that prevented multiplayer from ever loading (when i was trying to customize the "connecting..." page).

You're saying you don't get the intro logo, but you see the menus just fine? Although I tried playing with a friend and I think he had a similar issue where the menus worked, but as soon as he got into a game everything was black, though he could still hear the sounds and on my screen he was moving just fine. If your page reloads, that seems to be a different issue.

**Comment by schwarzjr** [comment]
Yes, I see only the menu. After starting a single player game, the page reloads in a couple of seconds with a Safari notice that a problem occured with this page so it was reloaded. If you have any ideas how I can help to pinpoint the issue, let me know.

**Comment by d07RiV** [comment]
Ah, that's a little different. If the whole browser tab crashes it becomes a lot harder to find the actual problem. Though if you're saying it runs on one of previous versions, that narrows down the possibilities. If you can run it locally, maybe you could find the exact commit that caused the issue? Or I can try to host multiple builds if you can't do that.

**Comment by schwarzjr** [comment]
What I know is that it starts crashing with the f0ba5c7 commit in gh-pages. What I don't know is which commits from the master branch it includes. But maybe you can find out easily. I think it was the 916bbd90cb57e2c5a3a50303782ae26a5315ea52 commit, but I am not sure.

**Comment by zzVoldemart** [comment]
9/1/19 i just downloaded this for the first time. I have the correct MPQ file but same thing. I see no logo but then I see the menu but as soon as I click on my character to play, I see the touch buttons but no game play. I’m on an iPhone X and a iPhone 6 s + and neither work. Help help

**Comment by zzVoldemart** [comment]
@D07RiV

**Comment by AJenbo** [comment]
Since there are so many issues with running it in Safari, have you tried the native iOS port?
https://github.com/diasurgical/devilutionX/pull/182

**Comment by zzVoldemart** [comment]
Ajenbo do you have a working copy of diablo on your phone and if so how do you run it ?

**Comment by zzVoldemart** [comment]
@AJenbo

**Comment by AJenbo** [comment]
Install it via Xcode, run it like any other app.

@zzVoldemart if you forget to @ someone, just edit your previous message

**Comment by zzVoldemart** [comment]
@AJenbo what’s Xcode I’m on my iPhone...

**Comment by AJenbo** [comment]
Xcode is apples developer tool for iPhone

**Comment by schwarzjr** [comment]
@AJenbo Yeah, I know. But the web version has some touch-related tweaks included which make iphone playing much more pleasant.

Btw had time to play with the code a bit and the strangest thing is that I can't seem to be able to build a working jscc/wasm files for iphone. They work just fine in Safari on macOS, but they don't on ios. The screen just stays black.

What I tried specifically: I check out a diabloweb commit I know works on ios (8c33947d), npm install, npm run build, npm run deploy - works on ios. In the next step I build (node build.js) what should be the same jscc/wasm files, replace them in the /src/api folder, again install, build and deploy - and it doesn't work on ios (screen is black) but works just fine on macOS safari.

Given the fact the jscc files differ sligtly (my build vs @d07RiV's build), I suspect that the problem may originate in different versions of emscripten, or something like that. Which version exactly do you use? I'd like to try the exact same settings to find out what is going on.

**Comment by schwarzjr** [comment]
Writing the last post got me thinking and ... Heureka! I found it! It is the -O3 optimization switch! When I compile it with -O0, it works.

**Comment by AJenbo** [comment]
Yeah that's fair enough, hopefully we can have an even better ui implementation for the native port in the not to distant future. We do have some concepts going but we need to work a bit more on the render for it to handle it and that won't happen before Hellfire lands.

**Comment by zzVoldemart** [comment]
@schwarzjr can you give me a working copy? I’m not very Tech savvy on the phone. I have the MPQ file but it’s not working like you describe. I can’t see any game play

**Comment by schwarzjr** [comment]
@zzVoldemart You can try it at https://schwarzjr.github.io/diabloweb/
But as I said - it crashes when going from the cathedral back to the town. But maybe you will be lucky :)

**Comment by AJenbo** [comment]
I'm going to guess the issue here is related to some of the old code that has since been cleaned up upstream
